### ISSUE TITLE
Security Rule support for Service Tags

### ISSUE THREAD

**Issue by paulschwarzenberger** [Azure/azure-sdk-for-go#2775]
### Feature Request

Support for service tags in security rules for network security groups, see screenshot for example configuration using the Azure portal. Looking at lines 15559 onwards in the[ network API](https://github.com/Azure/azure-sdk-for-go/blob/master/services/network/mgmt/2018-07-01/network/models.go) this does not appear to currently be supported.
![image](https://user-images.githubusercontent.com/26303407/45825170-889a1000-bce9-11e8-8a41-b15e70c8824c.png)

**Comment by paulschwarzenberger** [comment]
I raised this due to an[ issue with Terraform](https://github.com/terraform-providers/terraform-provider-azurerm/issues/1629). Following further testing, I have resolved the issue by setting service tags using Terraform such as Sql.WestEurope as a string, whereas tags such as Internet, VirtualNetwork are set as a list.

Assuming this is all supported by the Azure SDK for Go, it would be useful to update the documentation to include.
